src/Data/CodeGen.purs

```diff
@@ -151,23 +152,28 @@ writeCodeFromAbi opts abiPath destFile = do
       Left (AbiDecodeError err) -> do
         tell [ ABIError { abiPath, error: err.error, idx: err.idx } ]
         pure Nothing
-      Right res -> pure $ Just $ Identity $ res
-    let abi = map Identity $ maybeAnnotateArity $ map (\(Identity a) -> a) $ catMaybes abiUnAnn
+      Right res -> pure $ Just $ Identity res
+    let abi = map Identity $ maybeAnnotateArity $ un Identity <$> catMaybes abiUnAnn
     genCode (Abi $ abi) { exprPrefix: opts.exprPrefix, indentationLevel: 0 }
       # runImported opts destFile
       # writeTextFile UTF8 destFile
       # liftAff
 
 maybeAnnotateArity :: Array AbiType -> Array AbiType
-maybeAnnotateArity abi = evalState (traverse go abi) S.empty
+maybeAnnotateArity abi =
+  let
+    Tuple nonFuncAbi funcAbi = foldMap groupingFunc abi
```

Comment:
side effect of this is that we would have all generated code of functions in the end of file

Reply:
I'm ok with that